public/js/actions/pay-methods.js

```diff
@@ -0,0 +1,73 @@
+import $ from 'jquery';
+import braintree from 'braintree-web';
+
+import * as actionTypes from 'constants/action-types';
+import * as appActions from './app';
+
+
+export function getPayMethods(jquery=$) {
+  // Note: not currently used as payMethods
+  // are added to user state upon login.
+  return dispatch => {
+    jquery.ajax({
+      method: 'get',
+      url: '/api/braintree/mozilla/paymethod/',
+      context: this,
+    }).then(function(data) {
+      dispatch({
+        type: actionTypes.GOT_PAY_METHODS,
+        user: {
+          payMethods: data,
+        },
```

Comment:
When dispatching an action, I think it makes sense to keep it completely separate from the data layer. How about dispatching just `{type: ..., payMethods: data}` ? That way, the `user` part is not coupled to this at all.